src/views/open-api-docs-view/utils/group-operations.ts

```diff
@@ -0,0 +1,35 @@
+import { OperationProps, OperationGroup } from '../types'
+import { truncateHcpOperationPath } from '../utils'
+
+/**
+ * Given a flat array of operation prop objects,
+ * Return an array of operation groups
+ *
+ * Operation groups are constructed by organizing the incoming
+ * operation prop objects by their paths.
+ */
+export function groupOperations(
+	operationObjects: OperationProps[]
+): OperationGroup[] {
+	const operationGroupsMap = operationObjects.reduce(
+		(
+			acc: Record<string, { heading: string; items: OperationProps[] }>,
+			o: OperationProps
+		) => {
+			const truncatedPath = truncateHcpOperationPath(o._placeholder.__path)
+			const firstPathPart = truncatedPath.split('/').slice(0, 3).join('/')
```

Comment:
As a breadcrumb, could you give an example of an expected path here? Or rather what is the part that we are cutting out?

Reply:
Great call! I added a detailed example to the description comment for this function! For posterity, will mirror part of it here:

---

We start with the raw paths from the API spec. For HCP services, these have a long common prefix relating to the specific HCP service name, API version, organization, and project:

```yaml
ListApps: /secrets/2023-06-13/organizations/{organization_id}/projects/{project_id}/apps
CreateApp: /secrets/2023-06-13/organizations/{organization_id}/projects/{project_id}/apps
GetApp: /secrets/2023-06-13/organizations/{organization_id}/projects/{project_id}/apps/{app_name}
GetAppSecret: /secrets/2023-06-13/organizations/{organization_id}/projects/{project_id}/apps/{app_name}/secrets/{secret_name}
CreateAwsSmSyncIntegration: /secrets/2023-06-13/organizations/{organization_id}/projects/{project_id}/sync/aws-sm
```

We remove this common prefix with `truncateHcpOperationPath`:

```yaml
ListApps: /apps
CreateApp: /apps
GetApp: /apps/{app_name}
GetAppSecret: /apps/{app_name}/secrets/{secret_name}
CreateAwsSmSyncIntegration: /sync/aws-sm
```

Then we group the operations based on a slug created from their first two path segments. The slug for `ListApps` and `CreateApps` is `/apps`, so they're grouped together, and the slug for `GetApp` is `/apps/{app_name}`. So we end up with groups like:

```json
{
    "/apps": [ "ListApps", "CreateApp" ],
    "/apps/{app_name}": [ "GetApp", "GetAppSecret" ],
    "/sync/aws-sm": [ "CreateAwsSmSyncIntegration" ]
}
```